web-components/src/components/coachmark/Coachmark.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (c) Cisco Systems, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { FocusTrapMixin } from "@/mixins";
+import reset from "@/wc_scss/reset.scss";
+import styles from "./scss/module.scss";
+import { customElement, html, LitElement, property, query, PropertyValues } from "lit-element";
+import { classMap } from "lit-html/directives/class-map";
+import "@/components/button/Button";
+
+export type coachmarkPlacement = "auto" | "left" | "right" | "top" | "bottom";
+
+@customElement("md-coachmark")
+export class Coachmark extends FocusTrapMixin(LitElement) {
+  @property({ type: String }) message = "";
+  @property({ type: String }) actionname = "Next"
+  @property({ type: Boolean }) hidebutton = false;
+  @property({ type: String }) placement: coachmarkPlacement = "auto";
+  @property({ type: Boolean }) show = false;
+  @property({ type: String }) color = "default";
+
+  @query(".md-coachmark__popper") popper!: HTMLDivElement;
+  @query(".md-coachmark__reference") reference!: HTMLDivElement;
+
+  private slotContent: Element[] | null = null;
+
+  connectedCallback() {
+    super.connectedCallback();
+  }
+
+  notifyCoachCreate() {
+    this.dispatchEvent(
+      new CustomEvent("coach-create", {
+        bubbles: true,
+        composed: true,
+        detail: {
+          placement: this.placement,
+          reference: this.reference,
+          popper: this.popper
+        }
+      })
+    );
+  }
+
+  coachAction() {
+    this.dispatchEvent(
+      new CustomEvent("coach-action", {
+        bubbles: true,
+        composed: true,
+        detail: {
+          popper: this.popper
+        }
+      })
+    );
+  }
+
+  handleSlotChange(event: Event) {
+    const slot = event.target as HTMLSlotElement;
+    if (slot) {
+      const slotContent = slot.assignedElements({ flatten: true });
+      if (slotContent.length) {
+        this.slotContent = slotContent;
+      }
+    }
+  }
+
+  protected update(changedProperties: PropertyValues) {
+    super.update(changedProperties);
+    if (changedProperties.has("show")) {
+      if (this.show) {
+        this.setFocusableElements!();
+        if (this.focusableElements && this.focusableElements.length) {
+          this.focusableElements[0].focus();
+        }
+      }
+    }
+  }
+
+  get coachWrapClassMap() {
+    return {
+      "md-coachmark--active": !!this.show
+    };
+  }
+
+  get coachClassMap() {
+    return {
+      [`md-coachmark__popper--${this.placement}`]: !!this.placement,
+      [`md-coachmark__popper--${this.color}`]: !!this.color
+    };
+  }
+
+  static get styles() {
+    return [reset, styles];
+  }
+
+  render() {
+    return html`
+      <div class="md-coachmark ${classMap(this.coachWrapClassMap)}">
+        <div class="md-coachmark__popper ${classMap(this.coachClassMap)}" tabindex="0">
+          <div class="md-coachmark__content" >
```

Comment:
Doesn't seem like you are actually using popper.js here?

Reply:
Not sure if we need here popper? If yes, I fix it.